### ISSUE TITLE
CLI transactions command cannot handle optional columns in CSV output

### ISSUE THREAD

**Issue by BroderPeters** [grindsa/dkb-robo#74]
While playing around with the CLI transactions command, I ran into the following error.

```bash
Traceback (most recent call last):
[truncated to relevant dkb-robo paths]
  File "/home/bp/src/dkb-flow/.venv/lib/python3.12/site-packages/dkb_robo/cli.py", line 160, in transactions
    ctx.obj["FORMAT"](transactions_list)
  File "/home/bp/src/dkb-flow/.venv/lib/python3.12/site-packages/dkb_robo/cli.py", line 240, in formatter
    writer.writerows(data)
  File "/usr/lib/python3.12/csv.py", line 167, in writerows
    return self.writer.writerows(map(self._dict_to_list, rowdicts))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/csv.py", line 159, in _dict_to_list
    raise ValueError("dict contains fields not in fieldnames: "
ValueError: dict contains fields not in fieldnames: 'mandatereference'
```
I executed with this command:
```bash
dkb -u "<login>" --format csv transactions --account "<account_id>" --date-from 2024-12-01
```

The issue comes from [how the CSV field names are set](https://github.com/grindsa/dkb-robo/blob/20002337d7c09676c4b2057e57982a3f3ee0c678/dkb_robo/cli.py#L238). As they are set by the first data entry, later entries might cause the above error if the first line was missing optional fields.

Using version 0.28.1.

**Comment by grindsa** [comment]
Hi, I pushed a potential fix into a [separate branch](https://github.com/grindsa/dkb-robo/tree/cli_csv). Please check and let me know if it works for you.

**Comment by BroderPeters** [comment]
The code change avoids the error, but will also truncate the result to the columns from the first data entry.
In this particular case:  Omitting "mandatreference"

```bash
(.venv) ➜  dkb-robo git:(cli_csv) dkb -u "<login>" --format csv transactions --account "<account_id>" --date-from 2024-12-01
Password: 
check your banking app on "<device>" and confirm login...
Status: processing. Waiting for confirmation
Status: processing. Waiting for confirmation
amount,currencycode,date,bdate,vdate,postingtext,peeraccount,peerbic,peer,peerid
```

One strategy to ensure that it includes all field names from the data object could be to fetch the dict with most keys and utilize that as field names builder. I've visualized my thought [here](https://github.com/grindsa/dkb-robo/pull/75). That would result into something like this:

```bash
(.venv) ➜  dkb-robo git:(cli_csv) ✗ dkb -u "<login>" --format csv transactions --account "<account_id>" --date-from 2024-12-01
Password: 
check your banking app on "<device>" and confirm login...
Status: processing. Waiting for confirmation
amount,currencycode,date,bdate,vdate,customerreference,mandatereference,postingtext,reasonforpayment,peeraccount,peerbic,peer,peerid,text
```

Although, I think the proper way would be to consistently return a fixed set of all field names to make the command usage predictable. (But it's a bit late for me and I barely know the full code base to make a proper suggestion just now...)

**Comment by grindsa** [comment]
Thank you for the PR. You are right. The assumption to build the field names based on the dataset with the most keys should be good enough. The dictionary gets created by the library, so we have full control on the inserted keys (we could even specify the fieldnames manually).

**Comment by grindsa** [comment]
fix got included in [v0.28.2](https://github.com/grindsa/dkb-robo/releases/tag/0.28.2)
